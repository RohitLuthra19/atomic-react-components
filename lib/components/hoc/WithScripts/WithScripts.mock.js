export const defaultConfig = [
  {
    src: 'https://maps.googleapis.com/maps/api/js',
    async: true,
    scriptId: 'maps',
  },
  {
    src: 'https://pay.google.com/gp/p/js/pay.js',
    scriptId: 'pay',
  },
];

export const defaultConfig1 = [
  {
    src: 'https://maps.googleapis.com/maps/api/js',
    async: true,
    scriptId: 'maps',
  },
  {
    src: 'https://pay.google.com/gp/p/js/pay.js',
    scriptId: 'pay',
  },
];
