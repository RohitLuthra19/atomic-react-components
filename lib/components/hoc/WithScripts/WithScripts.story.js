import React from 'react';
import { storiesOf } from '@storybook/react';
import { defaultConfig, defaultConfig1 } from './WithScripts.mock';

// Import Styled Component to showcase variations
import withScripts from './WithScripts';

const comp1 = () => (
  <div>Hoc will consume the wrapper Component A and injected respective script details</div>
);
const comp2 = () => (
  <div>Hoc will consume the wrapper Component B and injected respective script details</div>
);

const ComponentA = withScripts(comp1, defaultConfig);

const ComponentB = withScripts(comp2, defaultConfig1);

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('remove duplicate script injection', module)
  .addParameters({ jest: ['ScriptInjection', 'ScriptInjectionVanilla'] })
  .addWithChapters('ScriptInjection', {
    chapters: [
      {
        sections: [
          {
            sectionFn: () => <ComponentA {...defaultConfig} />,
            options: {
              showSource: true,
              allowSourceToggling: true,
              showPropTables: true,
              allowPropTablesToggling: true,
            },
          },
        ],
      },
      {
        title: 'Primary ScriptInjection Variations',
        sections: [
          {
            title: 'Variation Name',
            sectionFn: () => <ComponentB {...defaultConfig1} />,
          },
        ],
      },
    ],
  });
