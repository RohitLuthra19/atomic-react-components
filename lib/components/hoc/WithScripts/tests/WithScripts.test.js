import { mount } from 'enzyme';

import withScripts from '../index';

import { defaultConfig } from '../WithScripts.mock';

const comp1 = () => (
  <div>Hoc will consume the wrapper Component A and injected respective script details</div>
);

const ComponentA = withScripts(comp1, defaultConfig);

describe('<WithScripts />', () => {
  test('should render correctly', () => {
    const WithScriptsComponent = mount(<ComponentA />);
    expect(WithScriptsComponent).toMatchSnapshot();
  });
});
