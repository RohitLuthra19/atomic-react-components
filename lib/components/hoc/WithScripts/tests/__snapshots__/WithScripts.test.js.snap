// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithScripts /> should render correctly 1`] = `
<WithScripts(comp1)>
  <comp1
    injectedProp={Object {}}
  >
    <div>
      Hoc will consume the wrapper Component A and injected respective script details
    </div>
  </comp1>
</WithScripts(comp1)>
`;
